ext/mysqlnd/mysqlnd_charset.c

```diff
@@ -479,260 +525,292 @@ static unsigned int my_ismbchar_gb18030(const char * start, const char * end)
 #define UTF8_MB4 "utf8mb4"
 #define UTF8_MB3 "utf8"
 
+static zend_always_inline unsigned int mysqlnd_mbcharlen_null(const unsigned int c)
+{
+	ZEND_UNREACHABLE();
+	return 0;
+}
+
+static zend_always_inline unsigned int check_null(const char * const start, const char * const end)
+{
+	ZEND_UNREACHABLE();
+	return 0;
+}
+
+static unsigned int mysqlnd_mbcharlen_dispatch(enum mysqlnd_encoding_charlen encoding, const unsigned int c)
+{
+	switch (encoding) {
+#define ENUMERATOR_DISPATCH(x) case x##_id: return x(c);
+		ENUMERATE_ENCODINGS_CHARLEN(ENUMERATOR_DISPATCH)
+#undef ENUMERATOR_DISPATCH
+		default: return mysqlnd_mbcharlen_null(c);
```

Comment:
Doesn't this appear twice in this `switch`? In the first entry in enum and in the default clause?

Reply:
We also need a default return so that the function returns something on all paths (otherwise it can be undefined behaviour). As such, I just put the null case as the default as well. All of this is optimized by the compiler so the duplicate case doesn't matter performance-wise.